src/framework/ui/button/button.component.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import {
+  TouchableOpacity,
+  TouchableOpacityProps,
+  GestureResponderEvent,
+  StyleSheet,
+  ImageSourcePropType,
+  Image,
+  ImageProps,
+} from 'react-native';
+import {
+  StyledComponentProps,
+  StyleType,
+  Interaction,
+  styled,
+} from '@kitten/theme';
+import {
+  Text as TextComponent,
+  Props as TextProps,
+} from '../text/text.component';
+import {
+  ButtonAlignment,
+  ButtonAlignments,
+} from './type';
+
+interface ButtonProps {
+  icon?: ImageSourcePropType;
+  status?: string;
+  size?: string;
+  alignment?: string | ButtonAlignment;
+  children?: React.ReactText;
+}
+
+const Text = styled<TextComponent, TextProps>(TextComponent);
```

Comment:
Why not import already styled component?

Reply:
This will cause require cycle. This is actual for each complex component